.github/workflows/README.md

```diff
@@ -0,0 +1,57 @@
+# Automaton flows on repository
+
+The Terrarium repo has various automation flows in place. Mainly there are three events that trigger the automation in this repo:
+
+1. New Pull request
+1. Merge to the base branch (main)
+1. Create new release tag (v*)
+
+## 1. New pull request
+
+The intent of automation pipelines here is to validate the changes coming from the pull requests.
+
+```mermaid
+flowchart LR
+  Code --> PR --> CH["Checks\nGitHub Action"]
+  PR --> CLI["CLI test automation\nCode Pipes"]
+  PR --> API["API test automation\nAZP"]
+  CH --> UT["Unit Test"]
+  CH --> ST["Smoke test\nAPI health-check & harvest flow"]
+```
+
+## 2. Merge to the base branch (main)
+
+There are two automation intents that run on merge to master:
+
+### 1. Validation
+
+This is same as the PR checks, but is important here to ensure sanity of the code merge.
+
+### 2. Build container images and deploy to staging
+
+The terrarium repo produces two docker images:
+
+- **docker.io/cldcvr/terrarium-api** - Terrarium API image is used to deploy the API server. It is based on alpine and contains the go binary for server.
+- **docker.io/cldcvr/terrarium-farm-harvester** - Terrarium Harvester image is based on terraform image and contains Terrarium CLI binary built in along with a Makefile that contains sequence of running the terraform harvest commands.
+
+```mermaid
+flowchart LR
+  Code --> Main --> H["Push harvester image\nGitHub Action\ncldcvr/terrarium-farm-harvester:latest,\ncldcvr/terrarium-farm-harvester:commit_sha"]
+  H --> F["Used in farm repo"]
+  Main --> API["Push API Image\nCode Pipes App Integration\ncldcvr/terrarium-api:latest,\ncldcvr/terrarium-api:commit_sha"]
+  API --> D["Deploy to Staging"]
+  Main --> C["Unit tests and integration test pipelines"]
+```
+
+## 3. Create new release tag (v*)
+
+The release process currently deals with releasing the new version of Terrarium CLI binary. The Terrarium API doesn't have a Prod release (or deployment) setup currently.
```

Comment:
It'd be nice to add the [homebrew-tap](https://github.com/cldcvr/homebrew-tap) related info as well. As this step will create a PR as well.
Further, we do have a TODO item as well, we need to use some cldcvr common user as author to this PR.
(https://github.com/cldcvr/terrarium/blob/640b4dd33566e3699cd15ffc7a6a1b09d7cca504/src/cli/terrarium/.goreleaser.yaml#L55)

IMO It'd be nice to record it here for now.

Reply:
Thanks @cc-shekher, I forgot the homebrew-tap part while writing this, adding it now.

The commit author part is something I'm not confident about. IFAIK GitHub doesn't let you create bot accounts, and even if we manage to create one, it'd be hard to figure who is taking ownership of this then. So IMO we should put the owner's details in the commit_author.

Reply:
@cc-shekher this is ready for review again.